module/zfs/zfs_vnops.c

```diff
@@ -399,7 +399,7 @@ zfs_write(znode_t *zp, uio_t *uio, int ioflag, cred_t *cr)
 			 */
 			woff = zp->z_size;
 		}
-		uio->uio_loffset = woff;
+		zfs_uio_offset(uio) = woff;
```

Comment:
This won't work for macOS, since you have to call `uio_setoffset()` ->
`zfs_uio_setoffset(uio, woff);`

Reply:
Assignment through the macros doesn't feel good to me either. `zfs_uio_setoffset(uio, woff)` looks like a cleaner approach.

Reply:
I agree. I went ahead and added another commit that addresses this. So, this patch should now work out for FreeBSD, macOS, and Linux.